src/data_structures/fenwick.md

```diff
@@ -29,138 +29,110 @@ It was first described in a paper titled "A new data structure for cumulative fr
     return res
 
 def increase(int i, int delta):
-    for all j with g(j) <= i <= j:
+    for all j with g(j) < i <= j:
         t[j] += delta
 ```
 
 The function `sum` works as follows:
 
-1. First, it adds the sum of the range $[g(r), r]$ (i.e. $T[r]$) to the `result`.
-2. Then, it "jumps" to the range $[g(g(r)-1), g(r)-1]$ and adds this range's sum to the `result`.
-3. This continues until it "jumps" from $[0, g(g( \dots g(r)-1 \dots -1)-1)]$ to $[g(-1), -1]$; this is where the `sum` function stops jumping.
+1. First, it adds the sum of the range $(g(r), r]$ (i.e. $T[r]$) to the `result`.
+2. Then, it "jumps" to the range $(g(g(r)), g(r)]$ and adds this range's sum to the `result`.
+3. This continues until it "jumps" from $[1, g(g(g(r) \dots ))]$ to $(g(0), 0]$; this is where the `sum` function stops jumping.
 
 The function `increase` works with the same analogy, but it "jumps" in the direction of increasing indices:
 
-1. The sum for each range of the form $[g(j), j]$ which satisfies the condition $g(j) \le i \le j$ is increased by `delta`; that is, `t[j] += delta`.
+1. The sum for each range of the form $(g(j), j]$ which satisfies the condition $g(j) \lt i \le j$ is increased by `delta`; that is, `t[j] += delta`.
 Therefore, it updates all elements in $T$ that correspond to ranges in which $A_i$ lies.
 
 The complexity of both `sum` and `increase` depend on the function $g$.
-There are many ways to choose the function $g$ such that $0 \le g(i) \le i$ for all $i$.
-For instance, the function $g(i) = i$ works, which yields $T = A$ (in which case, the summation queries are slow).
-We could also take the function $g(i) = 0$.
-This would correspond to prefix sum arrays (in which case, finding the sum of the range $[0, i]$ will only take constant time; however, updates are slow).
+There are many ways to choose the function $g$ such that $0 \le g(i) \lt i$ for all $i$.
+For instance, the function $g(i) = i-1$ works, which yields $T = A$ (in which case, the summation queries are slow).
+We could also take the function $g(i) = 1$.
+This would correspond to prefix sum arrays (in which case, finding the sum of the range $[1, i]$ will only take constant time; however, updates are slow).
 The clever part of the algorithm for Fenwick trees is how it uses a special definition of the function $g$ which can handle both operations in $O(\log N)$ time.
 
 ### Definition of $g(i)$ { data-toc-label='Definition of <script type="math/tex">g(i)</script>' }
 
-The computation of $g(i)$ is defined using the following simple operation:
-we replace all trailing $1$ bits in the binary representation of $i$ with $0$ bits.
-
-In other words, if the least significant digit of $i$ in binary is $0$, then $g(i) = i$.
-And otherwise the least significant digit is a $1$, and we take this $1$ and all other trailing $1$s and flip them.
+The computation of $g(i)$ is defined as:
+toggling of the last set $1$ bit in the binary representation of $i$.
 
 For instance we get
```

Comment:
The mathjax renders weirdly in the preview https://gh.cp-algorithms.com/1376/data_structures/fenwick.html
I don't think we need two newlines per line.